app/models/payment_intent.rb

```diff
@@ -75,14 +76,15 @@ def update_status_and_charges(payment_account, api_intent, action_source, source
         end
       when PaymentIntent.wca_statuses[:created],
         PaymentIntent.wca_statuses[:pending]
-        # Reset by the gateway
         self.update!(
           confirmed_at: nil,
           confirmation_source: nil,
           canceled_at: nil,
           cancellation_source: nil,
           wca_status: updated_wca_status,
         )
+      when PaymentIntent.wca_statuses[:requires_capture]
+        yield api_intent if block_given?
```

Comment:
You're not yielding the correct information here. In the "successful" case above, we're querying the individual _charges_ and yielding them to the outside block. Now you're yielding the overarching payment intent, which is substantially different from yielding charges.

The outer block does not know which of the switch case branches of this method body yielded to it, so this will not behave correctly in most cases.